### ISSUE TITLE
[1.2.0] CLI graph_remove Doesn't work with regex

### ISSUE THREAD

**Issue by grotnicki** [Cacti/cacti#1899]
Hello,

I've been testing the new CLI for deleting graphs, and giving any argument to --graph-regex results in an error:

> php /usr/share/cacti/cli/remove_graphs.php --graph-template-id=13 --graph-regex=up --list
> There are 1 Graphs to Remove.  Use the --force option to remove these Graphs.
> ID      GraphName
> PHP Warning:  Invalid argument supplied for foreach() in /usr/share/cacti/cli/remove_graphs.php on line 262

Log:

> 2018/09/06 15:42:30 - CMDPHP SQL Backtrace: (/cli/remove_graphs.php: 242 db_fetch_assoc)(/lib/database.php: 400 db_fetch_assoc_prepared)(/lib/database.php: 414 db_execute_prepared)(/lib/database.php: 274 cacti_debug_backtrace)
> --
> 2018/09/06 15:42:30 - CMDPHP ERROR: A DB Row Failed!, Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')' at line 1

It works fine without givingany regex.

**Comment by netniV** [comment]
Add a line just before the db_fetch_assoc() that should like:

```php
print "SELECT gl.id, gtg.title_cache
        FROM graph_local AS gl
        INNER JOIN graph_templates_graph AS gtg
        ON gl.id=gtg.local_graph_id
        INNER JOIN host AS h
        ON h.id = gl.host_id
        $sql_where";
```
This will give you the SQL being used, copy that and paste it here.  The reason for the first issue is that the SQL is failing (which you've listed second).  The result will return false but the sizeof() sees the variable as populated so considers it's size to be 1.  I've submitted a patch for that but we will probably need to add a further one based off your SQL statement.

**Comment by grotnicki** [comment]
SELECT gl.id, gtg.title_cache
        FROM graph_local AS gl
        INNER JOIN graph_templates_graph AS gtg
        ON gl.id=gtg.local_graph_id
        INNER JOIN host AS h
        ON h.id = gl.host_id
        WHERE gl.id > 0 AND gl.graph_template_id IN (13) AND ()

**Comment by netniV** [comment]
OK so this is again the same kind of issue.  The code isn't checking that it has a list of regex's properly and ends up applying a blank list between the brackets.  I'll apply a fix for that too.

**Comment by netniV** [comment]
I have fixed this in the development branch.  For 1.1.38, you will need to apply the same fixes as 318caae and 57b7582 to your local version.

**Comment by cigamit** [comment]
Marking resolved.
